### DISCUSSION TITLE
How to download PDF that requires a button click to print

### DISCUSSION THREAD

**Discussion by misterpeople** [seleniumbase/SeleniumBase#3083] [Q&A] [answered]
Hey, on the website i scrape, I have a button that opens the PDF viewer in Chrome when clicked, there a workaround to do this without opening the viewer or something specially without automating the Chrome Viewer?  
Also is i possible to change the file name before saving it? thanks!

<img width="275" alt="print2" src="https://github.com/user-attachments/assets/06910810-1d15-45bf-9640-29016f8d73ad">

**Comment by mdmintz** [comment, marked as answer]
Set external_pdf to True:

https://github.com/seleniumbase/SeleniumBase/blob/70b05ff27587bffa30acfe8ed4c2a4fe657b3d90/seleniumbase/plugins/sb_manager.py#L80

**Reply by misterpeople** [reply]
Thanks, but how do i set it up?
with SB(uc=True, proxy=proxy_url, incognito=True, external_pdf=True) as sb:
I tried this and it still opened the pdf viewer by chrome, I tried also setting it up differently but again it did not work.
Do i need first to click on the print button? im using windows

**Reply by mdmintz** [reply]
None is the default value.
You have to set it to True.

**Reply by misterpeople** [reply]
I did, it still did not help it simply opens the chrome print view. Maybe its because im on Windows?
with SB(uc=True, proxy=proxy_url, incognito=True, external_pdf=True) as sb:

**Reply by mdmintz** [reply]
You have to click on the file link directly to download it automatically when using external_pdf=True. If that isn’t working, then not sure.

**Reply by misterpeople** [reply]
I click on the "print" button when external_pdf=True and the normal chrome print webview is showing, nothing else is happening, I guess its again bug with windows

**Reply by tigtol** [reply]
Hi,
The problem doesn't seem to be related to Windows.
I am using `with SB(external_pdf=True, uc=True, locale_code="en", headed=True) as sb:` on Linux and pdf are still displayed in the internal pdf viewer by Chrome.
